fix(server): increase person search limit

jrasm91 opened the pull request:
Increase the person search results from 20 to 500. In the future we can look at making this endpoint paginated if needed.

Fixes #8822

jrasm91 committed:
fix(server): increase person search limit

mertalev reviewed (commented):
1. On web, `maximumLengthSearchPeople` is still fixed at 20.
2. Web should only start searching at the second character to ease server load; it's likely to get 500 results with just one character.
3. Not a blocker for this PR, but the group by / count likely makes this much less responsive for large libraries. It'd be snappy at any library size if it didn't do that.

jrasm91 commented:
@mertalev  - thanks for the hint about (1), which I've changed to 1,000 now (aligns with the server). I've skipped (2) as it was not an easy fix given the current state of the web components, but I have implemented (3) which should at least make (2) less of an issue.

mertalev reviewed (approved):
Nice! Without the count stuff it's fast enough that (2) doesn't really matter.
